### ISSUE TITLE
put method not implemented

### ISSUE THREAD

**Issue by jeffmaury** [khuxtable/wagon-gitsite#11]
I am using wagon-gitsite to upload a P2 repository to my Github pages. In order to do it, i am using the wagon-maven-plugin with your wagon-gitsite.
It does not work because the method put is not implemented.
You can find the POM file at: https://gist.github.com/gists/1062771/download

**Comment by khuxtable** [comment]
Correct. I only implemented putDirectory, because that is what is required to deploy site documentation.

I will eventually implement put, but I have some other things to fix first.

**Comment by jeffmaury** [comment]
I tried a dirty patch: route put method to putDirectories as this one handles files. Files are uploaded but the hierarchy is skipped.
I will debug to get more information.

**Comment by khuxtable** [comment]
If you get something you like, submit a patch and I'll apply it and credit you. Though at the moment, I'm busy with work-for-pay.

**Comment by rgladwell** [comment]
I created a pull request for this change but didn't check if it maintains the directory structure:

https://github.com/khuxtable/wagon-gitsite/pull/15
